### PR TITLE
checkbashisms: 2.0.0.2 -> 2.21.1

### PR DESCRIPTION
**Pull request by KAction**
Project on sourceforge is long unmaintained, development moved to
Debian devtools.

 * Upgrade to version from latest released devtools
 * Install manual page
 * Install bash completion script.

I do not use Bash, so I am not sure whether bash completion works.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/checkbashisms/default.nix [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
+    substituteInPlace ./scripts/checkbashisms.pl \
+      --replace '###VERSION###' "$version"
+  '';
   installPhase = ''
-    install -D -m755 $src $out/bin/checkbashisms
+    mkdir -p $out/{bin,share/man/man1,share/bash-completion/completions}
+    install -m755 ./scripts/$pname.pl $out/bin/$pname
+    install -m644 ./scripts/$pname.1 $out/share/man/man1
+    install -m644 ./scripts/$pname.bash_completion \
+      $out/share/bash-completion/completions/$pname
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/checkbashisms/default.nix:22]: Please use installShellCompletion

File: pkgs/development/tools/misc/checkbashisms/default.nix [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
-
+  buildPhase = ''
+    substituteInPlace ./scripts/checkbashisms.pl \
+      --replace '###VERSION###' "$version"
+  '';
   installPhase = ''
-    install -D -m755 $src $out/bin/checkbashisms
+    mkdir -p $out/{bin,share/man/man1,share/bash-completion/completions}
+    install -m755 ./scripts/$pname.pl $out/bin/$pname
+    install -m644 ./scripts/$pname.1 $out/share/man/man1
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/checkbashisms/default.nix:20]: Please use installManPage

File: pkgs/development/tools/misc/checkbashisms/default.nix [hunk 3]
```diff
@@ -1,20 +1,25 @@
...
-  # The link returns directly the script. No need for unpacking
-  dontUnpack = true;
-
+  buildPhase = ''
+    substituteInPlace ./scripts/checkbashisms.pl \
+      --replace '###VERSION###' "$version"
+  '';
   installPhase = ''
-    install -D -m755 $src $out/bin/checkbashisms
+    mkdir -p $out/{bin,share/man/man1,share/bash-completion/completions}
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/checkbashisms/default.nix:18]:
```suggestion
    mkdir -p $out/bin
```

**Comment by KAction** [comment]
Applied. Thanks for the tips. I probably should re-read nixpkgs manual.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 119754` at bdb63fa0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>checkbashisms</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`buildPhase` should probably contain `runHook preBuild` and `runHook postBuild`.

Near pkgs/development/tools/misc/checkbashisms/default.nix:14:3:
```
   |
14 |   buildPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/development/tools/misc/checkbashisms/default.nix:27:5:
```
   |
27 |     license = lib.licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/tools/misc/checkbashisms/default.nix:24:3:
```
   |
24 |   meta = {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/tools/misc/checkbashisms/default.nix:18:3:
```
   |
18 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

**Pull request merged**